File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/UDS/DSWUDSRelationBaseEntity.ts
import UDSRelationType = require('App/Models/UDS/UDSRelationType');
import DSWUDSBaseEntity = require('App/Models/UDS/DSWUDSBaseEntity');

class DSWUDSRelationBaseEntity<T> extends DSWUDSBaseEntity {
    RelationType: UDSRelationType;
    Relation: T;
}

export = DSWUDSRelationBaseEntity;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/UDS/DSWUDSRelationBaseEntity.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.UDS
{
    public abstract class DSWUDSRelationBaseEntity<T> : DSWUDSBaseEntity, IDSWUDSRelationEntity<T>
         where T : DSWBaseEntity
    {
        protected DSWUDSRelationBaseEntity(Guid uniqueId) : base(uniqueId)
        {
        }

        #region [ Properties ]
        public UDSRelationType RelationType { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual T Relation { get; set; }
        #endregion

    }
}
